packages/application-shell/src/hooks/use-applications-menu/use-applications-menu.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import FetchApplicationsMenu from './fetch-applications-menu.graphql';
+
+const defaultApiUrl = window.location.origin;
+
+const defaultConfig = {
+  queryOptions: {},
+  skipRemoteQuery: false,
+  options: {},
+};
+
+const useApplicationsMenu = (config = {}) => {
+  const mergedConfig = { ...defaultConfig, ...config };
+
+  const [menu, setMenu] = React.useState(null);
+
+  // Memoize the function to get the local config, for local development
+  const getDevConfig = React.useCallback(() => {
+    const { __DEV_CONFIG__: devConfig } = mergedConfig.options;
+    if (!devConfig) {
+      throw new Error(
+        'In development mode, you need to pass `__DEV_CONFIG__` options to `useApplicationsMenu`.'
+      );
+    }
+    return devConfig;
+  }, [mergedConfig.options]);
+
+  // Trigger loading the menu from local file, for local development
+  React.useEffect(() => {
+    if (mergedConfig.skipRemoteQuery) {
+      const devConfig = getDevConfig();
+      if (devConfig.menuLoader) {
+        devConfig.menuLoader().then(setMenu);
+      }
+    }
+  }, [getDevConfig, mergedConfig.skipRemoteQuery]);
+
+  // Prepare to fetch config.
+  // We set up the apollo query as lazy, in order to execute it conditionally
+  // since hooks cannot be defined conditionally.
+  const environment = useApplicationContext(context => context.environment);
+  const [loadMenuData, { called, data }] = useLazyQuery(FetchApplicationsMenu, {
```

Comment:
This is nice!

Reply:
Nice! Stumbled over this for a minute. Should we maybe alias `called` to `hasApplicationsMenuQueryBeenCalled`. I don't know, would just've helped me. I think it's something which will freak me out with Apollo and hooks is that we'll have these super gernic `data` and `called` variable names around.

Reply:
Sure thing, we can alias them if you prefer so